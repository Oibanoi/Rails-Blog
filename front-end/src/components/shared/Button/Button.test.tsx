import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { BackToHomeButton, LogoutButton } from '.';
describe('Button Component', () => {
  it('Renders button back to home correctly', () => {
    const { container } = render(
      <Router>
        <BackToHomeButton />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });

  it('Renders button logout correctly', () => {
    const { container } = render(
      <Router>
        <LogoutButton />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });
});
