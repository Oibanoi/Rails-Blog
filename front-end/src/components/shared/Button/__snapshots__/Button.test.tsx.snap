// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renders button back to home correctly 1`] = `
<div>
  <a
    href="/"
  >
    <button
      class="ant-btn css-dev-only-do-not-override-pr0fja ant-btn-primary"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <div
          icon="[object Object]"
        />
      </span>
      <span>
        Về trang chủ
      </span>
    </button>
  </a>
</div>
`;

exports[`Button Component Renders button logout correctly 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-pr0fja ant-btn-default"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <div
        icon="[object Object]"
      />
    </span>
    <span>
      Đăng xuất
    </span>
  </button>
</div>
`;
